pygmt/datasets/load_remote_dataset.py

```diff
@@ -592,8 +592,10 @@ def _load_remote_dataset(
                 kind=dataset.kind, outgrid=None, vfname=voutgrd
             )
 
-    # Full path to the grid if not tiled grids.
-    source = which(fname, download="a") if not resinfo.tiled else None
+    # Full path to the grid
+    source: str | list = which(fname, download="a")
+    if resinfo.tiled:
+        source = sorted(source)[0]  # get first grid for tiled grids
     # Manually add source to xarray.DataArray encoding to make the GMT accessors work.
     if source:
         grid.encoding["source"] = source
```

Comment:
Could add `verbose="q"` to the `which` call here, and probably remove the `if source:` line, but doesn't really matter if it's all gonna go away after the refactor to use `xr.load_dataarray` in #3932.

Reply:
I prefer to have verbose="d" added and if source removed, so that this PR is fully functional on its own.